model/DataObject.php

```diff
@@ -2187,54 +2105,60 @@ public function manyManyComponent($component) {
 		foreach($classes as $class) {
```

Comment:
I feel the need to apologise for this method, even though I didn’t originally write it! The refactor looks good though